pseudomonas_analysis/nbconverted/Viz_E-GEOD-33245_patterns.py

```diff
@@ -23,181 +25,245 @@
+
 
+# ### 1v2 compared with 1v4
 
 # In[10]:
 
 
-cmap = sns.cubehelix_palette(start=2.8, rot=.1, as_cmap=True)
-sns.scatterplot(data=merged_df,
-                #data=merged_df[merged_df["max Z score"]<0.5],
-                x="Test statistic (Real)_grp_1v2",
-                y="Test statistic (Real)_grp_1v3",
-                hue="max Z score",
-                size="max Z score",
-                linewidth=0,
-                alpha=0.7,
-                palette=cmap)
-plt.plot([0,4],[0,4],"--k")
+process.plot_two_conditions(merged_1v2_1v4_df, "1v2", "1v4", "WT vs crc mutant", "WT vs crcZ mutant")
 
 
+# ### 1v3 compared with 1v4
+
 # In[11]:
 
 
-merged_df[merged_df['max Z score']>20]
+process.plot_two_conditions(merged_1v3_1v4_df, "1v3", "1v4", "WT vs cbrB mutant", "WT vs crcZ mutant")
 
 
-# In[12]:
+# **Takeaway:**
+# Genes with high specificity score (i.e. genes with a high absolute value z-score) are located in the off x-y regions, as expected since these off diagonal regions represent those genes that are more differentially expressed in one condition versus the other. This shows some promise for using z-score to distinguish between generic and specific genes and we can start looking more into these trends.
 
+# ## DEGs found using traditional criteria and using z-score
+# 
+# When performing DE analysis, this can return hundreds of genes using traditional criteria (FDR adjusted p-value < 0.05 and log2 fold change > 2). We want to see what genes are selected when we add z-score as an additional criteria to indicate genes that are specific to the pertubagen in question.
 
-merged_df.to_csv("merged_E-GEOD_1v2_1v3.tsv", sep="\t")
+# ### 1v2
 
+# In[12]:
 
-# **Takeaway:**
-# * A few specific genes (with very high z-score) in the off x-y regions as expected. This shows some promise for using z-score to distinguish between generic and specific genes and we can start looking more into these trends.
 
-# ## Create dataframe to compare trends
-# Here we create dataframes that Deb can look at to find *things of interest* as we start looking into how to use our computational predictions of generic and specific genes.
-# 
-# She would like a dataframe with 1v2 and 1v3, 1v3 and 1v4.
+(DEGs_1v2_df,
+ degs_1v2_traditional,
+ degs_1v2_specific,
+ degs_1v2_generic,
+ degs_1v2_intersect,
+ degs_1v2_intersect_generic,
+ degs_1v2_diff) = process.get_and_save_DEG_lists(merged_1v2s_df, '1v2')
+
 
 # In[13]:
 
 
-# Read data
-grp_1v2 = pd.read_csv(grp_1v2_file, sep="\t", header=0, index_col=0)
-grp_1v3 = pd.read_csv(grp_1v3_file, sep="\t", header=0, index_col=0)
-grp_1v4 = pd.read_csv(grp_1v4_file, sep="\t", header=0, index_col=0)
-
-grp_1v2_raw = pd.read_csv(grp_1v2_raw_file, sep="\t", header=0, index_col=0)
-grp_1v3_raw = pd.read_csv(grp_1v3_raw_file, sep="\t", header=0, index_col=0)
-grp_1v4_raw = pd.read_csv(grp_1v4_raw_file, sep="\t", header=0, index_col=0)
+process.plot_venn(degs_1v2_traditional, degs_1v2_specific, degs_1v2_generic)
```

Comment:
So would you interpret this as showing that the traditional DE results have more overlap with your specific genes than with generically DE genes? Is that what you expect here?

Reply:
I would interpret these findings to mean that using traditional DE criteria you find a mix of generic and specific genes. And applying a z-score cutoff will supply researchers with a reasonable sized list of DEGs that are most relevant to the perturbagen in question that they can follow up with.